tests/linear_model/test_glm.py

```diff
@@ -203,3 +220,79 @@ def test_logistic_predict_proba_shape():
     lr.fit(X, y)
     prob = lr.predict_proba(X)
     assert prob.shape == (100, 2)
+
+@pytest.mark.parametrize(
+    "reg_type,data_generator", 
+    [(LinearRegression, "single_chunk_regression"), 
+    (LogisticRegression, "single_chunk_classification"),
+    (PoissonRegression, "single_chunk_count_classification")
+    ]
+)
+def test_model_coef_dask_numpy(reg_type, data_generator, request):
+    """Tests that models return same coefficients and intercepts with array types"""
+    X, y = request.getfixturevalue(data_generator)
+    np_mod, da_mod = reg_type(fit_intercept=True), reg_type(fit_intercept=True)
+    da_mod.fit(X, y)
+    np_mod.fit(X.compute(), y.compute())
+    np.testing.assert_allclose(da_mod.coef_, np_mod.coef_)
+    np.testing.assert_allclose(da_mod.intercept_, np_mod.intercept_)
+
+@pytest.mark.parametrize(
+    "reg_type, skl_reg_type, skl_params, data_generator", 
+    [(LinearRegression, sklLinearRegression, {}, "single_chunk_regression"), 
+    (LogisticRegression, sklLogisticRegression, {'penalty' : 'none'}, "single_chunk_classification"),
+    (PoissonRegression, sklPoissonRegressor, {'alpha' : 0}, "single_chunk_count_classification")
+    ]
+)
+def test_model_against_sklearn(reg_type, skl_reg_type, skl_params, data_generator, request):
+    """
+    Tests accuracy of model predictions
+
+    Notes:
+        1. Dask ML 'newton' solver does not implement a penalty, only
```

Comment:
Why does `solver='newton'` have to be used? `penalty='l2'` is passed to the Dask-ML optimization but [is unused in Dask-GLM][2] [and Dask-ML][3].

I think testing linear models without regularization is fine. In that case, the `penalty` and `C` values can be ignored for `solver='newton'`.

[2]:https://github.com/dask/dask-glm/blob/1daf4c5de0f44eb9405a40afb5506d4d821d407f/dask_glm/algorithms.py#L143
[3]:https://github.com/dask/dask-ml/blob/1e811ce48951a45a54859aca387510abdc085722/dask_ml/linear_model/glm.py#L147-L148

Reply:
I think `N` should be much larger than `p`. `N / p == 5` seems reasonable. Having `N == p` sounds like a bad idea, especially without regularization and only 10 informative features.

I don't think tests should use a special dataset. I think changing the error tolerance is better, especially if the relative error is used as an error condition (I can't find the comment, but it's `rel_error = LA.norm(est - truth) / LA.norm(truth)`). Relative error is used as a convergence criteria in most textbooks, so I think it's fine to include it in the tests too.